src/Symfony/Component/Security/Core/User/ChainUserProvider.php

```diff
@@ -53,7 +54,13 @@ public function loadUserByUsername($username)
     {
         foreach ($this->providers as $provider) {
             try {
-                return $provider->loadUserByUsername($username);
+                $user = $provider->loadUserByUsername($username);
+
+                if (!$user instanceof UserInterface) {
+                    throw new LogicException(sprintf('%s should return an object implementing UserInterface or throw UsernameNotFoundException. "%s" given.', get_class($provider), gettype($user)));
```

Comment:
IMO, argument 3 for `sprintf()` should be `is_object($user) ? get_class($user) : gettype($user)`.

Reply:
done.